cmake/OpenCVCompilerOptimizations.cmake

```diff
@@ -339,6 +340,11 @@ elseif(ARM OR AARCH64)
     ocv_update(CPU_FP16_IMPLIES "NEON")
     set(CPU_BASELINE "NEON;FP16" CACHE STRING "${HELP_CPU_BASELINE}")
   endif()
+elseif(MIPS)
+  ocv_update(CPU_MSA_TEST_FILE "${OpenCV_SOURCE_DIR}/cmake/checks/cpu_msa.cpp")
+  ocv_update(CPU_KNOWN_OPTIMIZATIONS "MSA")
+  ocv_update(CPU_MSA_FLAGS_ON "")
```

Comment:
Why is it empty? It should contain "-mmsa" or something similar.

Reply:
This option seems not be used elsewhere, but it should be better to be defined as '-mmsa' for future use. I will modify it.

Reply:
This option is used, but it not obvious:

* there is a list of `CPU_KNOWN_OPTIMIZATIONS` which contain `MSA`
* cmake first tries to compile file `CPU_${OPT}_TEST_FILE` with option `CPU_${OPT}_FLAGS_ON`, where `OPT=MSA`
* if compilation succeeded and `MSA` is requested by user (via `CPU_BASELINE`), then these options will be added to compiler

Is the `-mmsa` option the only one required by implemented intrinsics? What about `-fhard-float` or others? We should move these options from toolchain file to CompilerOptimization.cmake.

Reply:
Yes. -mmsa  option is enough. When '-mmsa' is included, '-mfp64' and '-mhard-float' will be enabled by default.  '-mmsa' has the same effect as '-mmsa -mfp64 -mhard-float'.